uniprotkb-rest/src/test/java/uk/ac/ebi/uniprot/api/uniprotkb/controller/UniprotKBSearchControllerIT.java

```diff
@@ -759,10 +352,417 @@ public void searchForDeletedInactiveEntriesReturnItself() throws Exception {
                 .andExpect(jsonPath("$.results.*.inactiveReason.inactiveReasonType", contains("DELETED")));
     }
 
-    private String saveEntry() {
-        UniProtEntry entry = UniProtEntryMocker.create(UniProtEntryMocker.Type.SP);
-        String acc = entry.getPrimaryAccession().getValue();
-        storeManager.save(DataStoreManager.StoreType.UNIPROT, entry);
-        return acc;
+    @Override
+    protected void cleanEntries() {
+        storeManager.cleanSolr(DataStoreManager.StoreType.UNIPROT);
     }
+
+    @Override
+    protected MockMvc getMockMvc() {
+        return mockMvc;
+    }
+
+    @Override
+    protected String getSearchRequestPath() {
+        return SEARCH_RESOURCE;
+    }
+
+    @Override
+    protected int getDefaultPageSize() {
+        return 25;
+    }
+
+    @Override
+    protected List<SearchField> getAllSearchFields() {
+        return Arrays.asList(UniProtField.Search.values());
+    }
+
+    @Override
+    protected String getFieldValueForValidatedField(SearchField searchField) {
+        UniProtField.Search search = UniProtField.Search.valueOf(searchField.getName());
+        log.debug("FIELD_NAME: ", search.getName());
+        String value = "";
+        if (searchField.getName().startsWith("ftlen_")) {
+            value = "[* TO *]";
+        }
+        switch (search) {
```

Comment:
would it make sense to have constants for some of the possible values that are used in the tests? E.g., private static final String ACCESSION = "P21802"; which is used on line 301, 602, 641 and maybe other places? Same point could go for other values seen in this switch statement.

The point of the suggestion is really only to save time in future if someone wants to refactor something.

Reply:
done... tks